src/de/unescape.rs

```diff
@@ -0,0 +1,243 @@
+
+    /// A testing wrapper around unescape
+    fn ue(source: &[u8]) -> String {
+        unescape(source).unwrap()
+    }
+
+    #[test]
+    fn unescape_works() {
+        // Unchanged because no unescaping happens
+        assert_eq!(ue(b""), "".to_string());
+        assert_eq!(ue(b"a"), "a".to_string());
+        assert_eq!(ue(b"ab"), "ab".to_string());
+        assert_eq!(ue(b"abc"), "abc".to_string());
+        assert_eq!(ue(b"a/c"), "a/c".to_string());
+        assert_eq!(ue(b"\xF0\x9F\x91\x8F"), "👏".to_string()); // U+1F44F
+
+        // even number of backslashes
+        assert_eq!(ue(br#"\\"#), "\\".to_string());
+        assert_eq!(ue(br#"\\\\"#), "\\\\".to_string());
+        assert_eq!(ue(br#"\\\\\\"#), "\\\\\\".to_string());
+        assert_eq!(ue(br#"\\\\\\\\"#), "\\\\\\\\".to_string());
+
+        // The 8 short escape sequences \", \\, \/, \b, \f, \n, \r, \t (alone, start, end, middle)
+        assert_eq!(ue(br#"\""#), "\"".to_string());
+        assert_eq!(ue(br#"\\"#), "\\".to_string());
+        assert_eq!(ue(br#"\/"#), "/".to_string());
+        assert_eq!(ue(br#"\b"#), "\x08".to_string());
+        assert_eq!(ue(br#"\f"#), "\x0C".to_string());
+        assert_eq!(ue(br#"\n"#), "\n".to_string());
+        assert_eq!(ue(br#"\r"#), "\r".to_string());
+        assert_eq!(ue(br#"\t"#), "\t".to_string());
+        assert_eq!(ue(br#"\"abc"#), "\"abc".to_string());
+        assert_eq!(ue(br#"\\abc"#), "\\abc".to_string());
+        assert_eq!(ue(br#"\/abc"#), "/abc".to_string());
+        assert_eq!(ue(br#"\babc"#), "\x08abc".to_string());
+        assert_eq!(ue(br#"\fabc"#), "\x0Cabc".to_string());
+        assert_eq!(ue(br#"\nabc"#), "\nabc".to_string());
+        assert_eq!(ue(br#"\rabc"#), "\rabc".to_string());
+        assert_eq!(ue(br#"\tabc"#), "\tabc".to_string());
+        assert_eq!(ue(br#"abc\""#), "abc\"".to_string());
+        assert_eq!(ue(br#"abc\\"#), "abc\\".to_string());
+        assert_eq!(ue(br#"abc\/"#), "abc/".to_string());
+        assert_eq!(ue(br#"abc\b"#), "abc\x08".to_string());
+        assert_eq!(ue(br#"abc\f"#), "abc\x0C".to_string());
+        assert_eq!(ue(br#"abc\n"#), "abc\n".to_string());
+        assert_eq!(ue(br#"abc\r"#), "abc\r".to_string());
+        assert_eq!(ue(br#"abc\t"#), "abc\t".to_string());
+        assert_eq!(ue(br#"xy\"abc"#), "xy\"abc".to_string());
+        assert_eq!(ue(br#"xy\\abc"#), "xy\\abc".to_string());
+        assert_eq!(ue(br#"xy\/abc"#), "xy/abc".to_string());
+        assert_eq!(ue(br#"xy\babc"#), "xy\x08abc".to_string());
+        assert_eq!(ue(br#"xy\fabc"#), "xy\x0Cabc".to_string());
+        assert_eq!(ue(br#"xy\nabc"#), "xy\nabc".to_string());
+        assert_eq!(ue(br#"xy\rabc"#), "xy\rabc".to_string());
+        assert_eq!(ue(br#"xy\tabc"#), "xy\tabc".to_string());
+
+        // Short escape sequences mixed
+        assert_eq!(ue(br#" \" \" \" "#), " \" \" \" ".to_string());
+        assert_eq!(ue(br#" \t \n \r "#), " \t \n \r ".to_string());
```

Comment:
Then again, I guess this would be very hard to write without escapes. Maybe this is fine as it is. And a note of use of raw, escaped - showing that we work like standard rust unescaping.

Reply:
I see your point in the example above. But when you work on this, you start valuing the consistency of the test layout. We need the escape sequences on the right to be converted in plain strings. And IDE hightlights this difference nicely by the way, to it is easier to read this than on Github.

> showing that we work like standard rust unescaping

Not sure if I understand correctly. But JSON escaping is very different from Rust escaping. They share a small subset, but that's it. E.g. there is `\b` and `\f` in JSON but not Rust. Or `\xAB` `\0` and `\u{123}` in Rust but not in JSON.

Reply:
Fair enough. And I did not open in my IDE